### PR TITLE
* add more demos

### PR DESCRIPTION
**Pull request by zhijianma**
1. Update Alpaca-CoT En refined recipe and dataset  with simhash deduplication by Data-Juicer
2. Add two demos:  data_process_loop and tool_dataset_splitting_by_language.
3. Modify and fix some minor description of demos

### CODE REVIEW

File: README_ZH.md [hunk 1]
```diff
@@ -202,6 +202,7 @@ python xxx.py --config configs/demo/process.yaml --language_id_score_filter.lang
   * 编程代码 (例如 [TheStack](https://huggingface.co/datasets/bigcode/the-stack)) [[ModelScope](https://modelscope.cn/studios/Data-Juicer/process_code_data/summary)]
   * 中文指令数据 (例如 [Alpaca-CoT](https://huggingface.co/datasets/QingyiSi/Alpaca-CoT)) [[ModelScope](https://modelscope.cn/studios/Data-Juicer/sft_data_zh/summary)]
 * 工具池:
+  * 多语言分割数据集集 [[ModelScope](https://modelscope.cn/studios/Data-Juicer/tool_dataset_splitting_by_language/summary)]
```
Comment (HYLcool) [inline README_ZH.md:205]: 多了一个“集”，此外，”按语言分割数据集“感觉会更贴切

Comment (zhijianma) [reply README_ZH.md]: 已修复，并重命名为 ”按语言分割数据集“

Comment (pan-x-c) [inline README_ZH.md]: 多语言分割数据集集 -> 多语言分割数据集

File: demos/README.md [hunk 2]
```diff
@@ -28,10 +31,12 @@ streamlit run app.py
 - Tool quality classifier (`tool_quality_classifier`)
   - This demo provides 3 text quality classifier to score the dataset.
 
+- Tool dataset splitting by language (`tool_dataset_splitting_by_language`)
```
Comment (pan-x-c) [inline demos/README.md:34]: Tool dataset splitting by language -> Dataset splitting by language

File: demos/tool_dataset_splitting_by_language/dataset_splitting_by_language.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+         target_dir,
+         text_keys_to_load=None,
+         text_key_to_process='text',
+         suffixes=[],
+         num_proc=1):
+    """
+    Load dataset from the source directory, then apply language identification
+    using the operation filter called `LanguageIDScoreFilter`,
+    finally, split the dataset by language and save it.
+    :param src_dir: path thats store dataset directory
```
Comment (pan-x-c) [inline demos/tool_dataset_splitting_by_language/dataset_splitting_by_language.py:38]: path thats store dataset directory -> path to store the dataset

File: demos/tool_dataset_splitting_by_language/dataset_splitting_by_language.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    :param target_dir: path to store subset files(`jsonl` format)
+    :param text_key: key name of field that stores sample text, default "text:
```
Comment (pan-x-c) [inline demos/tool_dataset_splitting_by_language/dataset_splitting_by_language.py:40]: default "text: -> default "text"

File: demos/tool_dataset_splitting_by_language/dataset_splitting_by_language.py (hunk 3 above)
Comment (zhijianma) [reply demos/tool_dataset_splitting_by_language/dataset_splitting_by_language.py]: done

File: demos/README.md (hunk 2 above)
Comment (zhijianma) [reply demos/README.md]: done

File: demos/tool_dataset_splitting_by_language/dataset_splitting_by_language.py (hunk 4 above)
Comment (zhijianma) [reply demos/tool_dataset_splitting_by_language/dataset_splitting_by_language.py]: done

**Review by HYLcool — APPROVED** [review]
LGTM

**Pull request merged**